src/main/java/com/qunite/api/security/JwtService.java

```diff
@@ -3,32 +3,63 @@
 import com.auth0.jwt.JWT;
 import com.auth0.jwt.algorithms.Algorithm;
 import com.auth0.jwt.interfaces.DecodedJWT;
+import com.qunite.api.domain.TokenPair;
 import com.qunite.api.domain.User;
+import com.qunite.api.service.TokenService;
 import java.time.Instant;
+import java.time.temporal.ChronoUnit;
 import java.util.Optional;
+import java.util.UUID;
+import lombok.RequiredArgsConstructor;
 import org.springframework.beans.factory.annotation.Value;
-import org.springframework.stereotype.Component;
+import org.springframework.stereotype.Service;
 
-@Component
+@Service
+@RequiredArgsConstructor
 public class JwtService {
   @Value("${jwt.secret}")
   private String secret;
 
   @Value("${spring.application.name}")
   private String issuer;
 
-  public String createJwtToken(User user) {
-    return JWT.create()
-        .withSubject(user.getUsername())
-        .withIssuer(issuer)
-        .withIssuedAt(Instant.now())
-        .sign(Algorithm.HMAC256(secret));
+  @Value("${jwt.access-token-expiration-time}")
+  private Integer expirationTime;
+
+  private final TokenService tokenService;
+
+  public TokenPair createJwt(User user) {
+    String accessToken = generateJwt(
+        user, expirationTime, ChronoUnit.SECONDS, TokenType.ACCESS);
+    String refreshToken = generateJwt(
+        user, 7, ChronoUnit.DAYS, TokenType.REFRESH);
+
+    TokenPair tokenPair = new TokenPair();
+    tokenPair.setAccessToken(accessToken);
+    tokenPair.setRefreshToken(refreshToken);
+    tokenPair.setOwner(user);
+    return tokenService.create(tokenPair);
   }
 
-  public Optional<DecodedJWT> verifyAccessToken(String token) {
+  public Optional<DecodedJWT> verifyJwt(String token, TokenType tokenType) {
     return Optional.of(JWT.require(Algorithm.HMAC256(secret))
+            .withIssuer(issuer)
+            .withClaim("type", tokenType.getValue())
+            .build()
+            .verify(token))
+        .filter(decodedJWT -> tokenService.isTokenValid(decodedJWT.getToken()));
+  }
+
+  private String generateJwt(User user, int expirationTime, ChronoUnit expirationTimeUnit,
+                             TokenType type) {
+    return JWT.create()
+        .withSubject(user.getId().toString())
         .withIssuer(issuer)
-        .build()
-        .verify(token));
+        .withIssuedAt(Instant.now())
+        .withExpiresAt(Instant.now().plus(expirationTime, expirationTimeUnit))
+        .withClaim("type", type.getValue())
+        .withClaim("username", user.getUsername())
```

Comment:
Lets discuss the usage and profit of this claim, because IMHO it makes no sense if the user is going to be unauthnticated after changing his username.

Reply:
Discussion, will not delete it. It is used in JwtAuthorizationFilter to create UsernamePassword token.

![image](https://github.com/q-unite/qunite-api/assets/103951082/b14655e1-411d-4d56-84d7-2a4f6cc0107e)


If you want to replace principal name from username to id then go ahead, but it is the topic for another pr

Reply:
I thought we could use userRepository.findById here and remove username claim

Reply:
do you think extra bd request per each endpoint request is a good idea?

Reply:
I think extra db request is worth it since token would not be invalidated after changing username